apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -0,0 +1,446 @@
+    if (updateState.error !== null) {
+      return (
+        <DeviceActionError
+          device={device}
+          t={t}
+          errorName={updateState.error.name}
+          translationContext="FirmwareUpdate"
+        >
+          <Button
+            type="main"
+            outline={false}
+            onPress={quitUpdate}
+            mt={6}
+            alignSelf="stretch"
+          >
+            {t("FirmwareUpdate.quitUpdate")}
+          </Button>
+        </DeviceActionError>
+      );
+    }
+
+    switch (updateState.step) {
+      case "allowSecureChannelRequested":
+        return (
+          <AllowManager
+            device={device}
+            wording={t("DeviceAction.allowSecureConnection")}
+          />
+        );
+      case "installOsuDevicePermissionRequested":
+        return (
+          <ConfirmFirmwareUpdate
+            device={device}
+            oldFirmwareVersion={deviceInfo.seVersion ?? ""}
+            newFirmwareVersion={firmwareUpdateContext.final.name}
+            t={t}
+          />
+        );
+      case "installOsuDevicePermissionDenied":
+        return (
+          <FirmwareUpdateDenied
+            device={device}
+            newFirmwareVersion={firmwareUpdateContext.final.name}
+            onPressRestart={triggerUpdate}
+            onPressQuit={quitUpdate}
+            t={t}
+          />
+        );
+      case "installOsuDevicePermissionGranted":
+        if (!firmwareUpdateContext.shouldFlashMCU) {
+          return <FinishFirmwareUpdate device={device} t={t} />;
+        }
+        return undefined;
+      case "flashingMcu":
+        if (updateState.progress === 1) {
+          return <FinishFirmwareUpdate device={device} t={t} />;
+        }
+        return undefined;
+      default:
+        return undefined;
+    }
+  }, [
+    updateState.error,
+    updateState.step,
+    updateState.progress,
+    device,
+    deviceInfo.seVersion,
+    firmwareUpdateContext.final.name,
+    firmwareUpdateContext.shouldFlashMCU,
+    t,
+    triggerUpdate,
+    quitUpdate,
+  ]);
+
+  return (
+    <>
+      <Flex flex={1} justifyContent="space-between">
+        <Flex>
+          <Text variant="h4" ml={5}>
+            {t("FirmwareUpdate.updateDevice", { deviceName })}
+          </Text>
+          <VerticalStepper steps={steps} />
+        </Flex>
+        <Flex mx={6} mb={8}>
+          <Alert type="info" title={t("FirmwareUpdate.doNotLeaveLedgerLive")} />
+        </Flex>
+      </Flex>
+      <QueuedDrawer
+        isRequestingToBeOpened={Boolean(deviceInteractionDisplay)}
+        noCloseButton
+      >
+        {deviceInteractionDisplay}
+      </QueuedDrawer>
+      <QueuedDrawer isRequestingToBeOpened={isCloseWarningOpen} noCloseButton>
+        <CloseWarning
+          onPressContinue={() => setIsCloseWarningOpen(false)}
+          onPressQuit={quitUpdate}
+        />
+      </QueuedDrawer>
+    </>
+  );
+};
+
+const FirmwareUpdateScreen = () => {
```

Comment:
But wait: so we're removing directly all the current/existing version of the firmware update screen/component ? We don't use this version with a feature flag for now ?

Reply:
This is a very old screen that was never used actually, this isn't the old firmware update, it's dead code. The old firmware update is elsewhere.